pkg/pluginmanager_service/plugin_manager.go

```diff
@@ -56,6 +56,7 @@ type PluginManager struct {
 	// shutdown syncronozation
 	// do not start any plugins while shutting down
 	shutdownMut sync.Mutex
```

Comment:
so I think this is a good approach - I think I'd do it more like this, using a RWmutex: 

```
shutdownMut sync.RWMutex
shuttingDown    bool

...



// return whether the plugin manager is shutting down
func (m *PluginManager) isShuttingDown() bool {
	m.shutdownMut.RLock()
	defer m.shutdownMut.RUnlock()
	return m.shuttingDown
}
 ...

func (m *PluginManager) Shutdown(*pb.ShutdownRequest) (resp *pb.ShutdownResponse, err error) {
	m.shutdownMut.Lock()
        m.shuttingDown = true
	m.shutdownMut.Unlock(
        ...
}
```

Reply:
Applied your proposed changes!